DatadogSessionReplay/Tests/SessionReplayTests.swift

```diff
@@ -63,6 +63,7 @@ class SessionReplayTests: XCTestCase {
         // Then
         let sr = try XCTUnwrap(core.get(feature: SessionReplayFeature.self))
         XCTAssertEqual(sr.recordingCoordinator.sampler.samplingRate, 42)
+        XCTAssertFalse(config.isConfiguredWithNewApi)
```

Comment:
`isConfiguredWithNewApi` was added in #2030 to distinguish what API customers have used and only send the new or deprecated privacy levels in telemetry.

Reply:
/question As discussed with the team, we will report only new levels in telemetry so we can remove the `isConfiguredWithNewApi`. Do you want to do it as part of this PR?

Reply:
I'm planning to take care of this in a separate PR.

Reply:
I might keep it though, if we can indicate which API was used in telemetry.

Reply:
I removed it completely, and also removed the deprecated privacy level from the `SessionReplayConfiguration` protocol, as we don't rely on it anymore.